pkg/discovery/processor/cluster_processor.go

```diff
@@ -65,68 +71,107 @@ func (processor *ClusterProcessor) ConnectCluster() error {
 	return nil
 }
 
+// Perform a single node validation
+func (processor *ClusterProcessor) checkNodesWorker(work chan *v1.Node, done chan bool, index int) {
```

Comment:
IMO, we can add another input argument to the `checkNodeWorker`:
```golang
checkNodesWorker(work chan *v1.Node, stop chan struct{}, done chan bool, index int)
```
The worker will listen on both the `work` and `stop` channel. If it receives a signal from `stop` channel, then the worker should stop.   
In this way, we don't have to drain the `work` channel.

Reply:
That complicates the flow. You have several scenarios, and you have to create additional channels to handle them. Any common control channel will have to contain enough messages to shut down every worker. The flow gets interesting at that point. Not sure if the approach is easier.

Reply:
I agree with Michael here that we don't complicate our flow here.